cmd/symmecrypt/main.go

```diff
@@ -20,25 +21,70 @@ import (
 	"github.com/ovh/symmecrypt/keyloader"
 )
 
+const (
+	EncryptionKeyBase64Env = "ENCRYPTION_KEY_BASE64"
+)
+
 var (
 	DefaultCipher = aesgcm.CipherName
 	ciphers       = []string{aesgcm.CipherName, chacha20poly1305.CipherName, aespmacsiv.CipherName, hmac.CipherName}
 
-	app = kingpin.New("symmecrypt CLI tool", fmt.Sprintf("A command-line utility to generate symmecrypt encryption keys. Available ciphers: %s (default: %s)", strings.Join(ciphers, ", "), DefaultCipher))
+	app = kingpin.New("symmecrypt", fmt.Sprintf(`A command-line utility tied to the symmecrypt library (https://github.com/ovh/symmecrypt). Generate new keys, and encrypt/decrypt arbitrary data.
+	Available ciphers: %s (default: %s).
+	For encrypt/decrypt, encryption keys are expected as a comma-separated list of base64-encoded keys in the environment variable ENCRYPTION_KEY_BASE64.
+	Identifiers can optionally be specified to choose which key to use, and in case several revisions of the same key identifier are present, normal fallback/priority rules are applied using timestamps, as specified in https://github.com/ovh/symmecrypt
 
-	newEncryption           = app.Command("new", "Generate a new random encryption key")
-	newEncryptionIdentifier = newEncryption.Arg("identifier", "identifier for the key").Required().String()
-	newEncryptionCipher     = newEncryption.Arg("cipher", "cipher for the key").Default(DefaultCipher).Enum(ciphers...)
+Example:
+	$ export ENCRYPTION_KEY_BASE64=$(symmecrypt new foobar aes-gcm --base64)
+	$ symmecrypt encrypt <<EOF >test.encrypted
+	foo
+	bar
+	baz
+	EOF
+	$ cat -e test.encrypted
+	^^JDM-1^EM-$M-^K1nX;^WM-^HC6^Xw^?^BM-.M-p^[M-%%=^M-^ZM-uM-%%M-2^H6M-sM-NM-FM-^H^RM-]g^_&$
+	$ symmecrypt decrypt <test.encrypted
+	foo
+	bar
+	baz
+`, strings.Join(ciphers, ", "), DefaultCipher))
+	useBase64 = app.Flag("base64", "Use base64 encoding for input/output.").Default("false").Bool()
 
-	encrypt       = app.Command("encrypt", "Encrypt data with selected key + cipher. Outputs hex.")
-	encryptCipher = encrypt.Arg("cipher", "cipher to use").Default(DefaultCipher).Enum(ciphers...)
+	newEncryption           = app.Command("new", "Generate a new random encryption key.")
+	newEncryptionIdentifier = newEncryption.Arg("identifier", "Identifier for the key").Required().String()
+	newEncryptionCipher     = newEncryption.Arg("cipher", "Cipher for the key").Default(DefaultCipher).Enum(ciphers...)
 
-	decrypt       = app.Command("decrypt", "Decrypt data with selected key + cipher. Input is hex.")
-	decryptCipher = decrypt.Arg("cipher", "cipher to use").Default(DefaultCipher).Enum(ciphers...)
+	encrypt           = app.Command("encrypt", "Encrypt arbitrary data from STDIN. The output will be the encrypted data.")
+	encryptIdentifier = encrypt.Arg("identifier", "Identifier of the encryption key to use").String()
```

Comment:
If i understand, this must be used when `ENCRYPTION_KEY_BASE64=<b64(foo)>,<b64(bar)>`; correct ?
Is it really useful ? i find it weird to keep several encryption key as an environment variable.

Reply:
yes, correct
re: passing several keys in env, one reason is to keep several revisions of the same key (same identifier, more recent), to allow backwards compatible decryption while encrypting with a new one. in such a case, the identifier is still not needed @ args.
having several keys (= several identifiers) may be a bit unnecessary indeed, but it seemed to facilitate integration in scripts (retrieve keys all at once, let the configuration manager @ symmecrypt handle it).